tools/api-key-rotation/api_key_rotation_checker/main.py

```diff
@@ -0,0 +1,346 @@
+                # so only legacy keys have it
+                if "createdBy" in apikey:
+                    # Create our API key object
+                    # if it has "createdBy"
+                    key_object = ApiKey(
+                        apikey["keyId"],
+                        apikey["displayName"],
+                        apikey["createdBy"],
+                        apikey["createTime"],
+                        project_id)
+                else:
+                    # Create our API key object
+                    # if it does NOT have "createdBy"
+                    key_object = ApiKey(
+                        apikey["keyId"],
+                        apikey["displayName"],
+                        "None",
+                        apikey["createTime"],
+                        project_id)
+
+                # We need to convert
+                # our creation time for comparison
+                converted_creation_date = time_convert(key_object)
+
+                # Extract API Key ID for logging
+                key_id = key_object.key_id
+
+                # If our key is older than x days (default 90)
+                # based on our compare_dates function
+                # add api key to appropriate variable container
+                logging.info("Checking API key: %s creation date.. \n", key_id)
+                # Convert to JSON for logging
+                key_object_json = key_object.to_json()
+                if compare_dates(converted_creation_date, rotation_date):
+                    keys_needing_rotation.append(key_object_json)
+                else:
+                    keys_not_needing_rotation.append(key_object_json)
+
+    # Format our API keys
+    keys_needing_rotation = "\n".join(keys_needing_rotation)
+    keys_not_needing_rotation = "\n".join(keys_not_needing_rotation)
+
+    # Once analysis is complete for all keys,
+    # log results
+    if keys_needing_rotation:
+        logging.warning(
+            "Found API keys older than %s days. Please rotate: \n%s \n", rotation_period, keys_needing_rotation) # pylint: disable = line-too-long
+
+    if keys_not_needing_rotation:
+        logging.info(
+            "The following API key(s) are not older than %s days: \n%s", rotation_period, keys_not_needing_rotation) # pylint: disable = line-too-long
+
+
+def create_token():
+    """
+    As of March 2021, Google has released a SDK for API keys
+    but the SDK is in a private alpha and not accessible for
+    public usage. You can view the SDK here:
+    https://cloud.google.com/sdk/gcloud/reference/alpha/services/api-keys
+
+    Therefore, we must use the requests library and make API calls
+    with a bearer token until the SDK is usable by the public.
+    """
+    access_token = subprocess.run(
+        "gcloud auth print-access-token",
+        shell=True,
+        check=True,
+        stdout=subprocess.PIPE,
+        universal_newlines=True)
+    token = access_token.stdout
```

Comment:
This is a sad excerpt because this method returns an access token for the current SDK user, and below the access token for the application is used, so we have two user accounts here.

Is this API not available for regular application and we need to authenticate as Cloud SDK?

Also, We have google-auth library which provides the kind of authorization you use here, so we can use the Google credentials SDK in both cases. https://github.com/googleapis/google-auth-library-python/blob/36e6f0feb41e901effc854a4f8c907deb9998f21/google/oauth2/credentials.py#L363

Reply:
> Is this API not available for regular application and we need to authenticate as Cloud SDK?

No it is not available at this time. API keys is in private alpha and there is no SDK available.